src/service/BlockService.ts

```diff
@@ -0,0 +1,87 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { sha3_256 } from 'js-sha3';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { BlockRepository } from '../infrastructure/BlockRepository';
+import { ReceiptRepository } from '../infrastructure/ReceiptRepository';
+import { MerklePathItem } from '../model/blockchain/MerklePathItem';
+import { UInt64 } from '../model/UInt64';
+
+/**
+ * Transaction Service
+ */
+export class BlockService {
+
+    /**
+     * Constructor
+     * @param blockRepository
+     * @param receiptRepository
+     */
+    constructor(private readonly blockRepository: BlockRepository, private readonly receiptRepository: ReceiptRepository) {
+    }
+
+    /**
+     * Validate transaction hash in block
+     * @param leaf transaction hash
+     * @param height block height
+     */
+    public validateTransactionInBlock(leaf: string, height: UInt64): Observable<boolean> {
+        const rootHashObservable = this.blockRepository.getBlockByHeight(height);
+        const merklePathItemObservable = this.blockRepository.getMerkleTransaction(height, leaf);
+        return combineLatest(rootHashObservable, merklePathItemObservable).pipe(
+            map((combined) => this.validateInBlock(leaf, combined[1].merklePath, combined[0].blockTransactionsHash)),
+        );
+    }
+
+    /**
+     * Validate receipt hash in block
+     * @param leaf receipt hash
+     * @param height block height
+     */
+    public validateReceiptInBlock(leaf: string, height: UInt64): Observable<boolean> {
+        const rootHashObservable = this.blockRepository.getBlockByHeight(height);
+        const merklePathItemObservable = this.receiptRepository.getMerkleReceipts(height, leaf);
+        return combineLatest(rootHashObservable, merklePathItemObservable).pipe(
+            map((combined) => this.validateInBlock(leaf, combined[1].merklePath, combined[0].blockReceiptsHash)),
+        );
+    }
+
+    /**
+     * @internal
+     * Validate leaf against merkle tree in block
+     * @param leaf Leaf hash in merkle tree
+     * @param merklePathItem Merkle path item array
+     * @param rootHash Block root hash
+     */
+    private validateInBlock(leaf: string, merklePathItem: MerklePathItem[] = [], rootHash: string): boolean {
+        if (merklePathItem.length === 0) {
+                return leaf.toUpperCase() === rootHash.toUpperCase();
+        }
+        const rootToCompare = merklePathItem.reduce((proofHash, pathItem) => {
+                const hasher = sha3_256.create();
+                // Left
+                if (pathItem.position === 1) {
+                    return hasher.update(Buffer.from(pathItem.hash + proofHash, 'hex')).hex();
+                } else {
+                    // Right
+                    return hasher.update(Buffer.from(proofHash + pathItem.hash, 'hex')).hex();
+                }
```

Comment:
The position is being changed from a number to a (left/right) string. Rest has done the change in master, we are still waiting for the open api PR https://github.com/nemtech/nem2-openapi/pull/77

@dgarcia360 can we move 77 PR forward? once released we can then generate the clients and update the ts and java PR related to merkle tests

Reply:
https://github.com/nemtech/nem2-openapi/releases/tag/v0.8.4